### DISCUSSION TITLE
Does `tokio` support stack coroutine like `go` ?

### DISCUSSION THREAD

**Discussion by loongs-zhang** [tokio-rs/tokio#4440] [Q&A] [answered]
Does `tokio` support stack coroutine like `go` ?

**Comment by Darksonn** [comment, marked as answer]
What do you mean by this? Async/await is coroutine-based internally.

**Reply by loongs-zhang** [reply]
> What do you mean by this? Async/await is coroutine-based internally.

When a worker thread is scheduled to an I/O task, will the worker thread be `park` if the task is not completed? If so, when the worker threads are limited (for example, the machine is `1-core`), it takes `1000` seconds to execute `1000` I/O tasks (assuming that each I/O task takes `1 second`);

Under the same conditions, using `goroutine` takes only about `1.1 seconds`.

**Reply by Darksonn** [reply]
Tokio is able to switch between multiple tasks on a single thread when they are waiting for IO. Check out [this article](https://ryhl.io/blog/async-what-is-blocking/).

**Reply by loongs-zhang** [reply]
When I want to write code to prove it, I have a problem. Can you offering some help ? https://github.com/tokio-rs/tokio/discussions/4442
